internal/cabi/cabi.go

```diff
@@ -613,3 +629,50 @@ func (p *Transformer) transformCallbackFunc(m llvm.Module, fn llvm.Value) (wrap
 	}
 	return wrapFunc, true
 }
+
+func (p *Transformer) callMemcpy(m llvm.Module, ctx llvm.Context, b llvm.Builder, dst llvm.Value, src llvm.Value, size int) llvm.Value {
+	memcpy := p.getMemcpy(m, ctx)
+	var sz llvm.Value
+	if p.is32Bit {
+		sz = llvm.ConstInt(ctx.Int32Type(), uint64(size), false)
+	} else {
+		sz = llvm.ConstInt(ctx.Int64Type(), uint64(size), false)
+	}
+	return b.CreateCall(memcpy.GlobalValueType(), memcpy, []llvm.Value{
+		dst, src, sz, llvm.ConstInt(ctx.Int1Type(), 0, false),
+	}, "")
+}
+
+const (
+	memcpy64 = "llvm.memcpy.p0i8.p0i8.i64"
+	memcpy32 = "llvm.memcpy.p0i8.p0i8.i32"
```

Comment:
Reference from https://llvm.org/docs/LangRef.html#llvm-memcpy-intrinsic :
```
This is an overloaded intrinsic. You can use llvm.memcpy on any integer bit width and for different address spaces. Not all targets support all bit widths however.
```

See simular usage:
https://github.com/goplus/llgo/blob/cf2d1ef9ca2f56bc6fda5d083d90507bfbd0863a/ssa/memory.go#L260-L264

Correct function type is needed.

Maybe `cFunc` is not a good usage, need a better intrinsic wrapper.

Reply:
fixed